templates/gitlab.rb.j2

```diff
@@ -70,5 +70,12 @@ nginx['ssl_verify_client'] = "{{ gitlab_nginx_ssl_verify_client }}"
 nginx['ssl_client_certificate'] = "{{ gitlab_nginx_ssl_client_certificate }}"
 {% endif %}
 
+# Extra configuration
+{% if gitlab_extra_settings is defined %}
+{%- for extra in gitlab_extra_settings -%}
+{{ extra.key }} = {{ extra.value }}
```

Comment:
You might want to do a type check here and add quotes for strings, no quotes for numbers.